modules/core/src/lib/layer.ts

```diff
@@ -1129,22 +1137,25 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
           terrainSkipRender
         };
 
+        const projectProps = {
+          viewport,
+          devicePixelRatio,
+          modelMatrix,
+          coordinateSystem,
+          coordinateOrigin
+        } as ProjectProps;
```

Comment:
If possible, we should avoid forced casts as they hide problems.
```suggestion
        } as const satisfies ProjectProps;
```